lib/src/map/flutter_map_state.dart

```diff
@@ -92,49 +130,98 @@ class FlutterMapState extends MapGestureMixin {
+                                        _createLayer(layer, options.plugins),
+                                  )
+                              ],
+                            ),
                           ],
                         ),
                       ),
                     ),
-                    Stack(
-                      children: [
-                        if (widget.nonRotatedChildren.isNotEmpty)
-                          ...widget.nonRotatedChildren,
-                        if (widget.nonRotatedLayers.isNotEmpty)
-                          ...widget.nonRotatedLayers.map(
-                            (layer) => _createLayer(layer, options.plugins),
-                          )
-                      ],
+                  )
+                : GestureDetector(
+                    onScaleStart: handleScaleStart,
+                    onScaleUpdate: handleScaleUpdate,
+                    onScaleEnd: handleScaleEnd,
+                    onTap: _positionedTapController.onTap,
+                    onLongPress: _positionedTapController.onLongPress,
+                    onTapDown: _positionedTapController.onTapDown,
+                    onTapUp: handleOnTapUp,
+                    child: ClipRect(
+                      child: Stack(
+                        children: [
+                          OverflowBox(
+                            minWidth: size.x as double?,
+                            maxWidth: size.x as double?,
+                            minHeight: size.y as double?,
+                            maxHeight: size.y as double?,
+                            child: Transform.rotate(
+                              angle: mapState.rotationRad,
+                              child: Stack(
+                                children: [
+                                  if (widget.children.isNotEmpty)
+                                    ...widget.children,
+                                  if (widget.layers.isNotEmpty)
+                                    ...widget.layers.map(
+                                      (layer) =>
+                                          _createLayer(layer, options.plugins),
+                                    )
+                                ],
+                              ),
+                            ),
+                          ),
+                          Stack(
+                            children: [
+                              if (widget.nonRotatedChildren.isNotEmpty)
+                                ...widget.nonRotatedChildren,
+                              if (widget.nonRotatedLayers.isNotEmpty)
+                                ...widget.nonRotatedLayers.map(
+                                  (layer) =>
+                                      _createLayer(layer, options.plugins),
+                                )
+                            ],
+                          ),
+                        ],
+                      ),
```

Comment:
This is just duplicated code, which needs to be moved to a new method that can be called from this builder method. This reduce technical dept and makes the code more maintainable. 

```dart
Widget _buildMap() { ... }

...
     return MapStateInheritedWidget(
        mapState: mapState,
        child: Listener(
            onTap: handleTap,
            onLongPress: handleLongPress,
            onDoubleTap: handleDoubleTap,
            child: options.allowPanningOnScrollingParent  
              ? GestureDetector(
                   onTap: _positionedTapController.onTap,
                   onLongPress: _positionedTapController.onLongPress,
                   onTapDown: _positionedTapController.onTapDown,
                   onTapUp: handleOnTapUp,
                   child: _buildMap(),
                 )
              :  GestureDetector(
                    onScaleStart: handleScaleStart,
                    onScaleUpdate: handleScaleUpdate,
                    onScaleEnd: handleScaleEnd,
                    onTap: _positionedTapController.onTap,
                    onLongPress: _positionedTapController.onLongPress,
                    onTapDown: _positionedTapController.onTapDown,
                    onTapUp: handleOnTapUp,
                    child: _buildMap())),
...
```

Reply:
sure. I will do this

Reply:
@kengu my latest commit should fix the code duplication issue. Please review and let me know if any further changes required.